### PR TITLE
Io/parsing

### PR DESCRIPTION
**Pull request by Failte21**
Création du module IO Parsing:
- Valide les inputs utilisateurs
- Affiche l'usage du programme

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,3 +20,10 @@ cabal.project.local
 cabal.project.local~
 .HTF/
 .ghc.environment.*
+
+
+# IDE
+.vscode
```
Comment (etrobert) [inline .gitignore]: On utilise tous, est ce qu'on ne versionnerait pas le .vscode/settings.json pour qu'on ait les memes parametres propres au projet ?

File: IOParsing.hs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+module IOParsing
+(
+    parseArgs,
+    usage
+) where
+
+import qualified Data.ByteString.Lazy as B (readFile, ByteString)
+
+data IOResult a = Usage | Help | IOError a
```
Comment (etrobert) [inline IOParsing.hs]: How is that an "IO" Result ?
How is that an "IO" Error ?

Comment (etrobert) [reply IOParsing.hs]: IMO it is more a ParseResult.
But then "Result" implies that it is the type that describes all the possibles outcomes
But here it is "Anything that is not a file and an input"

File: IOParsing.hs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+usage Usage = usageBase
+usage Help = usageVerbose
+usage (IOError error) = usageError error
+
+parseFlag :: String -> Either (IOResult String) (B.ByteString, String)
+parseFlag "-h" = Left Help
+parseFlag "--help" = Left Help
+parseFlag flag = Left $ IOError $ "Unkown flag " ++ flag
+
+parseArgs :: [String] -> IO (Either (IOResult String) (B.ByteString, String))
```
Comment (etrobert) [inline IOParsing.hs]: I don't think the extraction of the description file is enough to make "parseArgs" a function returning an IO
IMO it should be a pure function that just "parse the arguments" into a filename and an input, and then we can extract the file content

File: .gitignore (hunk 1 above)
Comment (Failte21) [reply .gitignore]: Il y a pas de linter en haskell ?

Comment (Failte21) [reply .gitignore]: Mais j'ai la flemme de regarder alors faisons au plus simple ^^

Comment (etrobert) [reply .gitignore]: https://hackage.haskell.org/package/hlint

**Review by etrobert — APPROVED** [review]

**Pull request merged**